Assignment 3-B

sciclic opened the pull request:
Hi @vincenzocoia and @almas2019, here is my draft of Assignment 3-B. I have designed the assignment with your notes from #81 in mind, therefore, primarily based on the [assignment as it was in 2018](https://github.com/UBC-STAT/stat545.stat.ubc.ca/blob/master/content/evaluation/hw08/hw08.md), with a few adaptations. Here are some notes / areas that I think still need some brainstorming / polishing:

1. **Assignment options and setting up:**
+ As suggested in #81, there are two options for the Assignment: modifying the BC Liquor app, or designing a new app from scratch. On this note, I am not sure of *where to find this BC Liquor app* or *how to push it to all of the student homework repos*, since you noted that they would be working on their homework repo in an assignment 3-B subfolder.
+ I am not even sure that we **should** push it to all their repos, I was thinking that perhaps we could have them confirm whether they would like to work on Option A or B beforehand, but again, not sure of how feasible this is logistically.
2. **Features:**
+ I have adjusted the feature suggestions from the 2018 version to be more general / applicable to both Option A and Option B, but kept some examples that are specific to Option A. Do we think what constitutes "a feature" is clear enough from these examples + the "What do we expect from features?" section in the rubric? Should an overarching statement be added?
+ I think that the rubric for creating the features could be improved, but I'm not sure of how to do so. The two requirements currently under Option A (there must be at least one functional widget, and one change to the UI) are from the 2018 version, but I wasn't sure of how to adapt those requirements to Option B seeing as they will be working from scratch. In a way, I think this may be covered in the *What do we expect from features?* section, but if you think more clarity is needed we can work on that. Are there any other notes that should be added on this section?
3. **Rubric:**
+ I have written the rubric thinking that it can double as a checklist that the students can use, and as guidance for the TAs in marking. Is this a good idea?
+ Are the requirements for the README too much?
+ Should we make the point about penalizing "nonsense" features (irrelevant to the app usage) more specific, perhaps with any examples?

Apologies for the lengthy text and I hope the assignment is at a good starting point, looking forward to your feedback & improving it!

Thank you,
Icíar

vincenzocoia commented:
This is organized very well, thank you. I only made a few minor changes -- the only notable one being that I removed all references to an external repo.

> I am not sure of where to find this BC Liquor app or how to push it to all of the student homework repos

Sorry about that -- [here's the source](https://github.com/daattali/shiny-server/tree/master/bcl). You can just copy the contents of Dean's `bcl` folder to an `assignment-3b` folder in the [assignment template repository](https://github.com/stat545ubc-2020/template-545b-assignments). **IMPORTANT**: You'll also need to copy over the [LICENSE](https://github.com/daattali/shiny-server/blob/master/LICENSE) file. For added transparency, I'd say to include a README in the `assignment-3b` folder that says that the BCL app was originally taken from Dean.

In fact, I have a feeling we'll be getting students wanting to tweak one of Dean's other shiny apps in his `shiny-server` repo. Do you think we should indicate that this is okay to do? We probably wouldn't even need to change the rubric. 

As for how to submit a pull request for the students -- Diana has written instructions [here](https://github.com/UBC-STAT/stat-545-instructor/tree/master/scripts). I've yet to try this myself -- if this is proving difficult, please do reach out to the team.

> I am not even sure that we should push it to all their repos

I've added instructions to delete the contents. Actually, even better, they should just _not_ accept the pull request! I just thought of that.

> 2. Features

This all looks okay to me.

> I have written the rubric thinking that it can double as a checklist

Effective

> Are the requirements for the README too much?

I don't think so.

> Should we make the point about penalizing "nonsense" features (irrelevant to the app usage) more specific, perhaps with any examples?

Perhaps. At least, maybe change the wording from a negative tone (penalizing nonsense features) to a positive tone (expecting sensible features). I think it might be useful to also add some context here: we just don't have enough time to cover how to make effective design decisions in class, but at the same time, we're still expecting the features you add to at least be sensible.

almas2019 commented:
@sciclic It looks good. Just a small thing to add, in the Readme when we talk about app description, do you think we should be more explicit in what we mean? Also I think we can add to the usefulness part (where we talk about whether it adds to the user experience), we can ask the students to maybe describe how each feature adds to the user experience (maybe a short sentence for each feature)?

almas2019 commented on content/assignments/assignment-3B.Rmd:
```diff
@@ -1,10 +1,120 @@
+    - [ ] Feature 3. (4 pts)
+  - *What are we looking for in your features?*
+     + Variety. Including three tables as part of your dashboard and having the same exact feature for each of them (e.g. a filtering option, even if the tables are distinct) will be penalized; at least 2 of the 3 features must be unique. 
+     + For either option A or B, we will evaluate the *usefulness* of your features in the context of your app (does it add to the user experience?). If we consider any of your features to be irrelevant for the app, 0.5pts will be penalized for that particular feature.
```
Specifically in this context. We can ask students write a sentence per feature to show how the feature is useful

sciclic replied:
Yeah you make a great point, I think we need to be more specific like this. I will add this! Thank you.

almas2019 commented on content/assignments/assignment-3B.Rmd:
```diff
@@ -1,10 +1,120 @@
+
+## Feature ideas 
+
+There are countless possibilities for features, but here are just a few ideas. You can choose any 3 or more ideas from this list or do anything else! Note that many of the examples in this list refer to the BC Liquor app, but they are all still applicable to creating your own Shiny app if you choose to do that instead. 
```
Maybe link to https://rstudio.github.io/shiny/tutorial/#welcome for more ideas for features

sciclic replied:
Added!

sciclic commented:
Thank you for your comments @almas2019 and @vincenzocoia, I implemented them accordingly - added some more clarity to the expectations for features, including a requirement for students to write one line per feature explaining its usefulness as per Almas' suggestion, and linking them to the shiny tutorial for more ideas for features (thank you for that!). I also changed the wording for the expectations regarding usefulness to be more positive (expecting sensible features vs. penalizing irrelevant ones), and added one line to the overview noting that the grading rubric doubles as a checklist and can be found at the end of the document.

I copied the following to the `assignment-3b` folder (that I created) within the [template assignment repo](https://github.com/stat545ubc-2020/template-545b-assignments):
+ Dean's `bcl` folder, which I renamed to `BCL-Shiny` for clarity. I added the LICENSE file within this folder. 
+ Created a README within the assignment-3b acknowledging Dean for the creation of the BCL app, and reiterating the instructions regarding what the README must contain. 
+ The `assignment-3b.rmd` file.

Hope this is okay! :)

sciclic commented:
I'm just going to merge this now, I closed it earlier by mistake and reopened it, but I think it's ready to go!
